visions/accessors.md

```diff
@@ -0,0 +1,348 @@
+
+This only leaves `consuming get` and `consuming read` as meaningful combinations.  A `consuming get` can be used for operations such as the `dropOldest` example above that model a transformation by creating a new value and terminating the old one.  The `consuming read` variant can be used similarly.  Unlike `borrow`, the `read` operation coroutine structure forces the containing value to live for a certain period of time, the value is consumed only at the end of that coroutine.
+
+## Some Observations
+
+Based on the above considerations, we can make a few useful observations about how these different accessors complement one another:
+
+#### Diversity is needed
+
+No one pair of accessors suffices to cover the needs of the language:
+
+* `get`/`set` require the property value to be copyable.
+* `unsafeAddress`/`unsafeMutableAddress` and `borrow`/`mutate` require the value to be already available in memory.  In particular, the mutating forms cannot be used to store a new value into a collection such as `Dictionary`, since the storage for such a value would not already be initialized.
+* `read`/`modify` exposes the value only for the lifetime of a coroutine, which is itself limited by implementation restrictions.
+
+#### Some combinations are nonsensical
+
+* Providing both a `mutate` and a `modify` for the same property doesn’t make sense.  If `mutate` is possible, then you can provide the value without making a copy, which means `modify` is unnecessary and redundant.
+
+#### We need to support existing APIs
+
+The current Swift API seems to require the following:
+
+* `set` is needed to be able to initialize new values for subscript operations.  `modify` is capable of doing this for types that can be trivially initialized, but there can be considerable overhead to construct an empty placeholder, expose it for mutation, and then record the final result.
+* Either `borrow` or `read` is needed to provide safe reading of noncopyable values
+* Either `mutate` or `modify` is needed to provide safe mutation of noncopyable values
+* We have `Dictionary` APIs (among others) that expose values as a different type than they are stored.  These can be supported using `read`/`modify` and storing the transformed value in the coroutine frame for the duration of the access.  They could also be supported using `get` to return a transformed value, but that would require a more expensive read-modify-write cycle for modifications.
+* Because `read`/`modify` have constraints on the lifetime of the access, they cannot fully replace `borrow`/`mutate`.
```

Comment:
Possibly related to the other comment: what is the difference between the lifetime constraints?

Reply:
The more detailed description further down for `read` and `borrow` illustrates this difference.